proxy/src/stream.rs

```diff
@@ -79,6 +106,25 @@ impl<S: AsyncWrite + Unpin> PqStream<S> {
         self.stream.flush().await?;
         Ok(self)
     }
+
+    /// Write the error message using [`Self::write_message`], then re-throw it.
+    /// Allowing string literals is safe under the assumption they might not contain any runtime info.
+    pub async fn throw_error_str<T>(&mut self, error: &'static str) -> anyhow::Result<T> {
```

Comment:
> It is used only for ERROR_INSECURE_CONNECTION, can we turn it into UserFacingError somehow?

That's another way to solve this, but it would require more code: we'd have to introduce a new proxy type and implement UserFacingError. So thumbs down from me, sorry :)

> If user facing error is implemented for everything that implements ToString, then str literal can be used without problems

That's a bad idea, because implementing UserFacingError for any ToString would mean that compiler wouldn't be able to warn us if we were to show an internal error to user. In fact, it would be possible to send `anyhow::Error`.